packages/react-components/react-dialog/src/components/Dialog/useDialog.ts

```diff
@@ -54,9 +51,19 @@ export const useDialog_unstable = (props: DialogProps): DialogState => {
     requestOpenChange,
   });
 
-  const handlebackdropClick = useEventCallback((event: React.MouseEvent<HTMLDivElement>) => {
+  const { disableBodyScroll, enableBodyScroll } = useDisableBodyScroll();
+
+  React.useEffect(() => {
```

Comment:
```suggestion
  useIsomorphicLayoutEffect(() => {
```

might be worth doing this before the browser paints